kbs/src/attestation/intel_trust_authority/mod.rs

```diff
@@ -3,17 +3,38 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use super::Attest;
+use crate::attestation::{generic_generate_challenge, make_nonce};
 use crate::token::{
     jwk::JwkAttestationTokenVerifier, AttestationTokenVerifier, AttestationTokenVerifierConfig,
     AttestationTokenVerifierType,
 };
 use anyhow::*;
 use async_trait::async_trait;
 use base64::{engine::general_purpose::STANDARD, Engine};
+use kbs_types::Challenge;
 use kbs_types::{Attestation, Tee};
 use reqwest::header::{ACCEPT, CONTENT_TYPE};
 use serde::{Deserialize, Serialize};
 use serde_json::json;
+use strum::{AsRefStr, Display, EnumString};
+
+const SUPPORTED_HASH_ALGORITHMS_JSON_KEY: &str = "supported-hash-algorithms";
+const SELECTED_HASH_ALGORITHM_JSON_KEY: &str = "selected-hash-algorithm";
+
+const ERR_NO_TEE_ALGOS: &str = "ITA: TEE does not support any hash algorithms";
+const ERR_INVALID_TEE: &str = "ITA: Unknown TEE specified";
```

Comment:
Let me do a refactoring about error information assertion later. Now it's good to me.